### PR TITLE
Added Quai ledger scope checks to all EVM state interactions

### PR DESCRIPTION
**Pull request by jdowning100**

### CODE REVIEW

File: common/address.go [hunk 1]
```diff
@@ -38,25 +38,54 @@ type AddressData interface {
...
 
 func (a Address) IsInQiLedgerScope() bool {
-	LedgerMask := byte(0b10000000)
 	// The first bit of the second byte is set if the address is in the Qi ledger
 	return a.Bytes()[1]&LedgerMask == LedgerMask
 }
 
+func (a Address) IsInQuaiLedgerScope() bool {
+	// The first bit of the second byte is not set if the address is in the Quai ledger
+	return a.Bytes()[1]&LedgerMask != LedgerMask
```
Comment (wizeguyy) [inline common/address.go:86]: this works, but IMO would be more readable to just check if `b[1] > 127`

**Review by wizeguyy — APPROVED** [review]

File: common/address.go (hunk 1 above)
Comment (jdowning100) [reply common/address.go:86]: if b[1] > 127 the address would be in the Qi ledger scope. Is that what you mean?

File: internal/quaiapi/api.go [hunk 2]
```diff
@@ -1562,6 +1562,13 @@ func (s *PublicTransactionPoolAPI) SendRawTransaction(ctx context.Context, input
 	if err != nil {
 		return common.Hash{}, err
 	}
+	// Set To address to Internal if it was accidentally unmarshalled as External (or vice-versa)
+	if tx.Type() != types.QiTxType {
```
Comment (gameofpointers) [inline internal/quaiapi/api.go:1566]: we dont need this since we use protodecode right?

Comment (jdowning100) [reply internal/quaiapi/api.go:1565]: This is an error check, the to could still be in the wrong scope

File: common/address.go (hunk 1 above)
Comment (jdowning100) [reply common/address.go]: I have updated the code as you requested

**Review by gameofpointers — APPROVED** [review]

**Review by wizeguyy — APPROVED** [review]

**Pull request merged**